proxy/envoy/config.go

```diff
@@ -163,6 +180,43 @@ func build(instances []*model.ServiceInstance, services []*model.Service,
 				},
 			}},
 		}
+
+		// Add TCP filters that share a port with HTTP filters.
+		if _, ok := tcpRouteConfigs[port]; ok {
+
+			// TCP and HTTP filter on the same port seems to cause
+			// envoy to assign all traffic to TCP filter regardless of
+			// the routing config. Temporarily drop TCP filters that
+			// share a port with HTTP (e.g. istio-egress) until this is
+			// root caused.
+
```

Comment:
To solve this, you need to use the cluster IP address of the service or range of IPs of the pods (latter is hard, former is s good approximation). I suggest adding a note with regard to this. And eliminate the comment below.